spec/words_counted/tokeniser_spec.rb

```diff
@@ -0,0 +1,139 @@
+          tokens = Tokeniser.new("Bayrūt").tokenise
+          expect(tokens).to eq(%w[bayrūt])
+        end
+      end
+
+      context "with `pattern` options" do
+        it "splits on accepts a custom pattern" do
+          tokens = Tokeniser.new("We-Are-ALL").tokenise(pattern: /[^-]+/)
+          expect(tokens).to eq(%w[we are all])
+        end
+      end
+
+      context "with `exclude` option" do
+        let(:tokeniser) { Tokeniser.new("That was magnificent, Trevor.") }
+
+        context "as a string" do
+          it "it accepts a string filter" do
+            tokens = tokeniser.tokenise(exclude: "magnificent")
+            expect(tokens).to eq(%w[that was trevor])
+          end
+
+          it "accepts a string filter with multiple space-delimited tokens" do
+            tokens = tokeniser.tokenise(exclude: "was magnificent")
+            expect(tokens).to eq(%w[that trevor])
+          end
+
+          it "normalises string filter" do
+            tokens = tokeniser.tokenise(exclude: "MAGNIFICENT")
+            expect(tokens).to eq(%w[that was trevor])
+          end
+        end
+
+        context "as a regular expression" do
+          it "filters on match" do
+            tokeniser = Tokeniser.new("That was magnificent, Trevor.")
+            tokens = tokeniser.tokenise(exclude: /magnificent/i)
+            expect(tokens).to eq(%w[that was trevor])
+          end
+        end
+
+        context "as a lambda" do
+          it "calls lambda" do
+            tokeniser = Tokeniser.new("That was magnificent, Trevor.")
+            tokens = tokeniser.tokenise(exclude: ->(token) { token.length < 5 })
+            expect(tokens).to eq(%w[magnificent trevor])
+          end
+
+          it "accepts a symbol for shorthand notation" do
+            tokeniser = Tokeniser.new("That was magnificent, محمد.}")
+            tokens = tokeniser.tokenise(exclude: :ascii_only?)
+            expect(tokens).to eq(%w[محمد])
+          end
+        end
+
+        context "as an array" do
+          let(:tokeniser) { Tokeniser.new("That was magnificent, Trevor.") }
+
+          it "accepts an array of strings" do
+            tokens = tokeniser.tokenise(exclude: ["magnificent"])
+            expect(tokens).to eq(%w[that was trevor])
+          end
+
+          it "accepts an array regular expressions" do
+            tokens = tokeniser.tokenise(exclude: [/that/, /was/])
+            expect(tokens).to eq(%w[magnificent trevor])
+          end
+
+          it "accepts an array of lambdas" do
+            filters = [
+              ->(token) {
+                token.length < 4
+              },
+              ->(token) {
+                token.length > 6
+              }
+            ]
+            tokens = tokeniser.tokenise(exclude: filters)
+            expect(tokens).to eq(%w[that trevor])
+          end
+
+          it "accepts a mixed array" do
+            filters = [
+              "that",
+              ->(token) {
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Use the `lambda` method for multi-line lambdas.